Helper/Cart.php

```diff
@@ -2790,4 +2779,63 @@ public function ignoreAdjustingShippingAmount($quote)
 
         return false;
     }
+    
+    /**
+     * Collect discount amount from each applied sale rules.
+     *
+     * @param \Magento\Quote\Model\Quote $quote
+     *
+     * @return array
+     */
+    public function getSaleRuleDiscounts($quote)
+    {
+        $address = $this->getCalculationAddress($quote);
+        if (!($appliedSaleRuleIds = $address->getAppliedRuleIds())) {
+            return [];
+        }
+        $salesRuleIds = explode(',', $appliedSaleRuleIds);
+        $saleRuleDiscountsDetails = [];
+        // If the Magento version < 2.3.4, we still collect discounts details via plugin methods.
+        if ($this->isCollectDiscountsByPlugin($quote)) {
+            $saleRuleDiscountsDetails = $this->sessionHelper->getCheckoutSession()->getBoltCollectSaleRuleDiscounts([]);
+        } else {
+            /* @var \Magento\SalesRule\Api\Data\RuleDiscountInterface $ruleDiscounts */
+            $extensionSaleRuleDiscounts = $address->getExtensionAttributes()->getDiscounts();
+            if ($extensionSaleRuleDiscounts && is_array($extensionSaleRuleDiscounts)) {
+                foreach ($extensionSaleRuleDiscounts as $value) {
+                    /* @var \Magento\SalesRule\Api\Data\DiscountDataInterface $discountData */
+                    $discountData = $value->getDiscountData();
+                    $saleRuleDiscountsDetails[$value->getRuleID()] = $discountData->getAmount();
+                }
+            }
+        }
+        
+        $ruleDiscountDetails = [];
+        foreach ($salesRuleIds as $salesRuleId) {
+            $rule = $this->ruleRepository->getById($salesRuleId);
+            // Only collect non-zero discount / free shipping promotion
+            if (!$rule || (!array_key_exists($salesRuleId, $saleRuleDiscountsDetails) && !$rule->getSimpleFreeShipping())) {
+                continue;
+            }
+            $ruleDiscountDetails[$salesRuleId] = array_key_exists($salesRuleId, $saleRuleDiscountsDetails) ? $saleRuleDiscountsDetails[$salesRuleId] : 0;
+        }
+        return $ruleDiscountDetails;
+    }
+    
+    /**
+     * If the Magento version < 2.3.4, we still collect discounts details via plugin methods.
+     *
+     * @param \Magento\Quote\Model\Quote $quote
+     *
+     * @return bool
+     */
+    public function isCollectDiscountsByPlugin($quote)
+    {
+        if ($this->configHelper->isActive($quote->getStore()->getId())
+            && version_compare($this->configHelper->getStoreVersion(), '2.3.4', '<')) {
+            return true;
+        }
+        
+        return false;
```

Comment:
**I think this can be refactored to:**
return $this->configHelper->isActive($quote->getStore()->getId()) && version_compare($this->configHelper->getStoreVersion(), '2.3.4', '<')

Reply:
Done